### ISSUE TITLE
Video quality control

### ISSUE THREAD

**Issue by k2arim99** [thiolliere/YouTubeStream#4]
I think there should a way to control the video quality of the stream it extracts

**Comment by thiolliere** [comment]
yes when opening a link we can have different stream available, opening a view offering the choice would be a huge improvement to me as well

**Comment by JackDesBwa** [comment]
+1!

I was about to propose something similar because I tried to see a VR180 video and it chose a low quality small monocular undistorted cropped version... not VR at all.
I studied the feasibility to get the URLs of these streams and found a partial way. Although it is not totally okay for me, the method lists the different versions for regular and 3D videos including spherical ones (but without the 3D part for the panoramic videos apparently), so it will be useful for this issue at least.

As you know with my previous contribution, I am not very interested in learning and moreover install the environment for compilation. However, I suppose that the "help wanted" flag you added to this issue concerns the algorithm to get the list, not to display it within the app.

So I tested my findings with a small script and it works pretty well (potential errors/exceptions not handled). It is written in python for simplicity of the quick test:
https://gist.github.com/JackDesBwa/bd0c07b091ada4344e5038a77c3b88e8

- The key is that the encoding is mixed between query string (url) format and json depending of the value.
- The usage of a "modern" user agent allows the server to list more video formats.
- I added a custom sort function to list higher definition videos first
- I added an example to retrieve the title, but other properties can be shown (including thumbnail, description...) from within the same downloaded document.
- I listed the streams titled with format, encoding and file size (whenever available), which are interesting information to display in my opinion. Maybe the width×height can be added too to complement when there is not a beautiful name to describe the definition.

I hope it will help for this functionality.
